crates/fuel-core/src/graphql_api/storage/balances.rs

```diff
@@ -0,0 +1,100 @@
+use fuel_core_storage::{
+    blueprint::plain::Plain,
+    codec::{
+        postcard::Postcard,
+        raw::Raw,
+    },
+    structured_storage::TableWithBlueprint,
+    Mappable,
+};
+use fuel_core_types::{
+    fuel_tx::{
+        Address,
+        AssetId,
+    },
+    fuel_vm::double_key,
+};
+use rand::{
+    distributions::Standard,
+    prelude::Distribution,
+    Rng,
+};
+
+pub type ItemAmount = u64;
+pub type TotalBalanceAmount = u128;
+
+double_key!(CoinBalancesKey, Address, address, AssetId, asset_id);
+impl Distribution<CoinBalancesKey> for Standard {
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> CoinBalancesKey {
+        let mut bytes = [0u8; CoinBalancesKey::LEN];
+        rng.fill_bytes(bytes.as_mut());
+        CoinBalancesKey::from_array(bytes)
+    }
+}
+
+impl core::fmt::Display for CoinBalancesKey {
+    fn fmt(&self, f: &mut core::fmt::Formatter) -> core::fmt::Result {
+        write!(f, "address={} asset_id={}", self.address(), self.asset_id())
+    }
+}
+
+/// This table stores the balances of coins per owner and asset id.
+pub struct CoinBalances;
+
+impl Mappable for CoinBalances {
+    type Key = CoinBalancesKey;
+    type OwnedKey = Self::Key;
+    type Value = TotalBalanceAmount;
+    type OwnedValue = Self::Value;
+}
+
+impl TableWithBlueprint for CoinBalances {
+    type Blueprint = Plain<Raw, Postcard>;
+    type Column = super::Column;
+
+    fn column() -> Self::Column {
+        Self::Column::CoinBalances
+    }
+}
+
+#[derive(Clone, Debug, Default, serde::Serialize, serde::Deserialize, PartialEq)]
+pub struct MessageBalance {
+    pub retryable: TotalBalanceAmount,
```

Comment:
This concept was already in `master`, however quite well hidden. See [here](https://github.com/FuelLabs/fuel-core/blob/master/crates/fuel-core/src/query/balance/asset_query.rs#L178) as the code is rejecting all messages that has non empty data.

I agree that the "name" retryable was indeed introduced in this PR. I'll figure out the exact difference and add documentation on the `Message` struct.

Reply:
@acerone85 
Here's more context on retryable vs. non-retryable messages.
1. original issue: https://github.com/FuelLabs/fuel-core/issues/946
2. PR in fuel-core: https://github.com/FuelLabs/fuel-core/pull/1067

In short, when message carries some additional (potentially valuable) data we don't want to spend it to pay for a transaction. IIRC, in case such transaction is reverted, the "potentially valuable" coin will be lost.